sklearn/cluster/_kmeans.py

```diff
@@ -969,6 +977,10 @@ def _init_centroids(
             If left to 'None' the number of centroids will be equal to
             number of clusters to form (self.n_clusters)
 
+        sample_weight : array-like of shape (n_samples,), default=None
```

Comment:
Since we are using `sample_weight.sum()`, we expect it to be a NumPy array.

```suggestion
        sample_weight : ndarray of shape (n_samples,), default=None
```